test/ts-scenario/steps/lib/listeners.ts

```diff
@@ -167,40 +176,43 @@ export async function checkListenerCallNumber(listenerName: string, compareNumbe
-				const msg: string = `Expected ${listenerName} to be called ${compareNumber} times, but was called ${gatewayListenerCalls} times`;
+				const msg = `Expected ${listenerName} to be called ${compareNumber} times, but was called ${gatewayListenerCalls} times`;
 				BaseUtils.logAndThrow(msg);
 			} else {
-				const msg: string = `Verified that the listener was called exactly ${compareNumber} times`;
+				const msg = `Verified that the listener was called exactly ${compareNumber} times`;
 				BaseUtils.logMsg(msg);
 			}
 			break;
 		case Constants.GREATER_THAN:
 			if (Number(gatewayListenerCalls) < Number(compareNumber)) {
 				throw new Error(`Expected ${listenerName} to be called a minimum ${compareNumber} times, but called ${gatewayListenerCalls} times`);
 			} else {
-				const msg: string = `Verified that the listener was called at least ${compareNumber} times`;
+				const msg = `Verified that the listener was called at least ${compareNumber} times`;
 				BaseUtils.logMsg(msg);
 			}
 			break;
 		case Constants.LESS_THAN:
-				if (Number(gatewayListenerCalls) > Number(compareNumber)) {
-					throw new Error(`Expected ${listenerName} to be called a maximum ${compareNumber} times, but called ${gatewayListenerCalls} times`);
-				} else {
-					const msg: string = `Verified that the listener was called a maximum ${compareNumber} times`;
-					BaseUtils.logMsg(msg);
-				}
-				break;
+			if (Number(gatewayListenerCalls) > Number(compareNumber)) {
+				throw new Error(`Expected ${listenerName} to be called a maximum ${compareNumber} times, but called ${gatewayListenerCalls} times`);
+			} else {
+				const msg = `Verified that the listener was called a maximum ${compareNumber} times`;
+				BaseUtils.logMsg(msg);
+			}
+			break;
 		default:
 			throw new Error(`Unknown condition type ${type} passed to checkListenerCallNumber()`);
 	}
 }
 
-export function checkContractListenerDetails(listenerName: string, listenerType: string, eventType: EventType, eventName: string, isActive: boolean): void {
+export function checkContractListenerDetails(listenerName: string, listenerType: string, eventType: EventType,
+	eventName: string, isActive: boolean): void {
 	const listenerObject: any = getListenerObject(listenerName);
 
 	// Check the listener properties
-	if ( (listenerObject.active !== isActive) || (listenerObject.type.localeCompare(listenerType) !== 0) || (listenerObject.eventName.localeCompare(eventName) !== 0) || (listenerObject.eventType !== eventType)) {
-		const msg: string = `Listener named ${listenerName} does not have the expected properties [type: ${listenerType}, eventName: ${eventName}, eventType: ${eventType}, active: ${isActive}]`;
+	if ((listenerObject.active !== isActive) || (listenerObject.type.localeCompare(listenerType) !== 0) ||
+	(listenerObject.eventName.localeCompare(eventName) !== 0) || (listenerObject.eventType !== eventType)) {
+		// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```

Comment:
Convert template variables to strings if possible. Should be able to use Node's `util.inspect()` to convert anything to a string if necessary